crypto/monero/address_marshal.go

```diff
@@ -0,0 +1,36 @@
+package mcrypto
+
+// MarshalText serializes the Monero Address type with some extra validation.
+func (a *Address) MarshalText() ([]byte, error) {
+	if err := a.validateDecoded(); err != nil {
+		return nil, err
+	}
+	return []byte(addrBytesToBase58(a.decoded[:])), nil
+}
+
+// UnmarshalText converts a base58 encoded monero address to our Address type.
+// The encoding, length and checksum are all validated, but not the network, as
+// it is unknown by the JSON parser. Empty strings are not allowed. Use an
+// address pointer in your serialized types if the Address is optional.
+func (a *Address) UnmarshalText(base58Input []byte) error {
```

Comment:
can you add a test for unmarshalling an address with the wrong length?

Reply:
There is a test:
```
func TestNewAddress_fail(t *testing.T) {
	_, err := NewAddress("fake", common.Mainnet)
	require.ErrorIs(t, err, errInvalidAddressLength)
}
```
We have full test coverage on the address specific code. The only lines that don't have coverage are unreachable code where I put in panics.